### ISSUE TITLE
[Bug]: [Nightly] Load failed showing "collection *** has not been loaded to memory or load failed"

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#18050]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 39ac588
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus==2.1.0.dev93
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Load failed showing "collection *** has not been loaded to memory or load failed"

```
[2022-07-04T15:51:54.596Z] [2022-07-04 15:07:24 - ERROR - pymilvus.decorators]: RPC error: [wait_for_loading_collection], <MilvusException: (code=1, message=collection search_collection_CQujbmSV has not been loaded to memory or load failed)>, <Time:{'RPC start': '2022-07-04 15:07:22.274972', 'RPC error': '2022-07-04 15:07:24.634754'}> (decorators.py:94)
[2022-07-04T15:51:54.596Z] [2022-07-04 15:07:24 - ERROR - pymilvus.decorators]: RPC error: [load_collection], <MilvusException: (code=1, message=collection search_collection_CQujbmSV has not been loaded to memory or load failed)>, <Time:{'RPC start': '2022-07-04 15:07:21.715764', 'RPC error': '2022-07-04 15:07:24.635045'}> (decorators.py:94)
[2022-07-04T15:51:54.596Z] [2022-07-04 15:07:24 - ERROR - ci_test]: Traceback (most recent call last):
[2022-07-04T15:51:54.596Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 22, in inner_wrapper
[2022-07-04T15:51:54.596Z]     res = func(*args, **kwargs)
[2022-07-04T15:51:54.596Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 53, in api_request
[2022-07-04T15:51:54.596Z]     return func(*arg, **kwargs)
[2022-07-04T15:51:54.596Z]   File "/usr/local/lib/python3.6/site-packages/pymilvus/orm/collection.py", line 469, in load
[2022-07-04T15:51:54.596Z]     conn.load_collection(self._name, replica_number=replica_number, timeout=timeout, **kwargs)
[2022-07-04T15:51:54.596Z]   File "/usr/local/lib/python3.6/site-packages/pymilvus/decorators.py", line 95, in handler
[2022-07-04T15:51:54.596Z]     raise e
[2022-07-04T15:51:54.596Z]   File "/usr/local/lib/python3.6/site-packages/pymilvus/decorators.py", line 91, in handler
[2022-07-04T15:51:54.596Z]     return func(*args, **kwargs)
[2022-07-04T15:51:54.596Z]   File "/usr/local/lib/python3.6/site-packages/pymilvus/decorators.py", line 73, in handler
[2022-07-04T15:51:54.596Z]     raise e
[2022-07-04T15:51:54.596Z]   File "/usr/local/lib/python3.6/site-packages/pymilvus/decorators.py", line 48, in handler
[2022-07-04T15:51:54.596Z]     return func(self, *args, **kwargs)
[2022-07-04T15:51:54.596Z]   File "/usr/local/lib/python3.6/site-packages/pymilvus/client/grpc_handler.py", line 673, in load_collection
[2022-07-04T15:51:54.596Z]     self.wait_for_loading_collection(collection_name, timeout)
[2022-07-04T15:51:54.596Z]   File "/usr/local/lib/python3.6/site-packages/pymilvus/decorators.py", line 95, in handler
[2022-07-04T15:51:54.597Z]     raise e
[2022-07-04T15:51:54.597Z]   File "/usr/local/lib/python3.6/site-packages/pymilvus/decorators.py", line 91, in handler
[2022-07-04T15:51:54.597Z]     return func(*args, **kwargs)
[2022-07-04T15:51:54.597Z]   File "/usr/local/lib/python3.6/site-packages/pymilvus/decorators.py", line 73, in handler
[2022-07-04T15:51:54.597Z]     raise e
[2022-07-04T15:51:54.597Z]   File "/usr/local/lib/python3.6/site-packages/pymilvus/decorators.py", line 48, in handler
[2022-07-04T15:51:54.597Z]     return func(self, *args, **kwargs)
[2022-07-04T15:51:54.597Z]   File "/usr/local/lib/python3.6/site-packages/pymilvus/client/grpc_handler.py", line 682, in wait_for_loading_collection
[2022-07-04T15:51:54.597Z]     return self._wait_for_loading_collection(collection_name, timeout)
[2022-07-04T15:51:54.597Z]   File "/usr/local/lib/python3.6/site-packages/pymilvus/client/grpc_handler.py", line 711, in _wait_for_loading_collection
[2022-07-04T15:51:54.597Z]     progress = self.get_collection_loading_progress(collection_name, timeout)
[2022-07-04T15:51:54.597Z]   File "/usr/local/lib/python3.6/site-packages/pymilvus/client/grpc_handler.py", line 690, in get_collection_loading_progress
[2022-07-04T15:51:54.597Z]     raise MilvusException(response.status.error_code, response.status.reason)
[2022-07-04T15:51:54.597Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=collection search_collection_CQujbmSV has not been loaded to memory or load failed)>
```

### Expected Behavior

load successfully

### Steps To Reproduce

```markdown
Running nightly:
https://ci.milvus.io:18080/jenkins/blue/rest/organizations/jenkins/pipelines/milvus-nightly-ci/branches/master/runs/633/nodes/190/steps/211/log/?start=0
```

### Milvus Log

1. Log: [artifacts-milvus-distributed-pulsar-master-633-pymilvus-e2e-logs.tar.gz](https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/633/artifact/artifacts-milvus-distributed-pulsar-master-633-pymilvus-e2e-logs.tar.gz)
2. FAILED time:
[2022-07-04T15:07:25.876Z] [gw3] [ 39%] FAILED testcases/test_search.py::TestCollectionSearch::test_search_normal_default_params[8-False-True-search_params0] 

3. collection name: search_collection_CQujbmSV

**Comment by yah01** [comment]
@congqixia concurrent write during iterating the SegmentsStatus (which is a map)

**Comment by yah01** [comment]
/assign @binbinlv 
fixed with #18054

**Comment by binbinlv** [comment]
This issue is fixed through the results of last nightly, but new issue #18088 occurs, I am wondering maybe there are some connections.

Close this  issue.
